apps/browser/qml/pages/components/HistoryItem.qml

```diff
@@ -26,9 +26,54 @@ ListItem {
 
     width: parent.width
     contentHeight: Math.max(Theme.itemSizeMedium, column.height + 2*Theme.paddingMedium)
+    menu: Component {
+        ContextMenu {
+            MenuItem {
+                //% "Open in normal mode tab"
+                text: webView.privateMode ? qsTrId("sailfish_browser-me-open-in-normal-mode-tab")
+                                          : //% "Open in private tab"
+                                            qsTrId("sailfish_browser-me-open-in-private-tab")
+                onClicked: view.load(model.url, model.title, true)
+            }
+            MenuItem {
+                //: Share link from browser history pulley menu
+                //% "Share"
+                text: qsTrId("sailfish_browser-me-share-link")
+                onClicked: webShareAction.shareLink(model.url, model.title)
+            }
+            MenuItem {
+                //% "Copy to clipboard"
+                text: qsTrId("sailfish_browser-me-copy-to-clipboard")
+                onClicked: {
+                    Clipboard.text = model.url
+                    //% "Link address copied"
+                    notification.text = qsTrId("sailfish_browser-me-link_address_copied")
+                    notification.show()
+                }
+            }
+            MenuItem {
+                //% "Add to bookmarks"
+                text: qsTrId("sailfish_browser-me-add-to-bookmarks")
+                onClicked: view.saveBookmark(model.url, model.title)
+            }
+            MenuItem {
+                //: Delete history entry
+                //% "Delete"
+                text: qsTrId("sailfish_browser-me-delete")
+                onClicked: historyDelegate.remove(model.url)
+            }
+        }
+    }
 
     ListView.onAdd: AddAnimation { target: root }
 
+    Notice {
+        id: notification
+
+        duration: Notice.Short
+        verticalOffset: -Theme.itemSizeMedium
+    }
+
     Row {
         id: row
         x: Theme.horizontalPageMargin
```

Comment:
On line 115, you'll need to add an extra `false` parameter:
```
    onClicked: view.load(model.url, model.title, false)
```
to avoid this error:
```
[W] unknown:115 - file:///usr/share/sailfish-browser/pages/components/HistoryItem.qml:115: Error: Insufficient arguments
```
Although you can skip parameters in javascript, apparently you can't for signals.

Reply:
Done